### PR TITLE
Changed `State-level ad targeting` at `brave://settings/rewards`

### PR DESCRIPTION
**Pull request by Tonev**
This changes `State-level ad targeting` to `Regional ad relevance` at brave://settings/rewards

Resolves https://github.com/brave/brave-browser/issues/22001

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: app/brave_generated_resources.grd [hunk 1]
```diff
@@ -398,7 +398,7 @@ Or change later at <ph name="SETTINGS_EXTENIONS_LINK">$2<ex>brave://settings/ext
         Maximum number of ads to display
       </message>
       <message name="IDS_SETTINGS_BRAVE_REWARDS_STATE_LEVEL_AD_TARGETING_LABEL" desc="The text label for the setting controlling subdivision-level ad targeting">
-        State-level ad targeting
+        Regional ad relevance
```
Comment (tmancey) [inline app/brave_generated_resources.grd:401]: Would be great if we could also update the description to help translators. Thanks

Comment (tmancey) [reply app/brave_generated_resources.grd:401]: Maybe see description from the brave://rewards copy

Comment (Tonev) [reply app/brave_generated_resources.grd:401]: @tmancey, what do you think about the following:

`The text label for the setting that allows Brave to show ads meant specifically for user's region`

Comment (Tonev) [reply app/brave_generated_resources.grd:401]: @tmancey, did you mean the message below, to make it match `brave://rewards` ?

`<message name="IDS_SETTINGS_BRAVE_REWARDS_STATE_LEVEL_AD_TARGETING_DESC_LABEL" desc="The text label for the description of subdivision-level ad targeting">
        This allows Brave to show you ads meant specifically for your state.`

Comment (tmancey) [reply app/brave_generated_resources.grd:401]: That is correct, thanks. And I do like "The text label for the setting that allows Brave to show ads meant specifically for user's region"

**Review by tmancey — APPROVED** [review]
LGTM :-)

**Comment by Tonev** [comment]
Thank you for your review, @tmancey! Please, merge whenever you think it is appropriate because I don't have required access to do it myself. Of course, if @aseren doesn't find anything that has to be fixed :+1:

**Review by tmancey — APPROVED** [review]
LGTM

**Pull request merged**